dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1994,6 +2040,305 @@ async fn cmd_db_region_missing_porst(
     Ok(())
 }
 
+/// List all regions
+async fn cmd_db_region_list(
+    datastore: &DataStore,
+    fetch_opts: &DbFetchOptions,
+    args: &RegionListArgs,
+) -> Result<(), anyhow::Error> {
+    use db::schema::region::dsl;
+
+    let regions: Vec<Region> = paginated(
+        dsl::region,
+        dsl::id,
+        &first_page::<dsl::id>(fetch_opts.fetch_limit),
+    )
+    .select(Region::as_select())
+    .load_async(&*datastore.pool_connection_for_tests().await?)
+    .await?;
+
+    check_limit(&regions, fetch_opts.fetch_limit, || {
+        String::from("listing regions")
+    });
+
+    if args.id_only {
+        for region in regions {
+            println!("{}", region.id());
+        }
+    } else {
+        #[derive(Tabled)]
+        struct RegionRow {
+            id: Uuid,
+            dataset_id: Uuid,
+            volume_id: Uuid,
+            block_size: i64,
+            blocks_per_extent: u64,
+            extent_count: u64,
+            read_only: bool,
+        }
+
+        let rows: Vec<_> = regions
+            .into_iter()
+            .map(|region: Region| RegionRow {
+                id: region.id(),
+                dataset_id: region.dataset_id(),
+                volume_id: region.volume_id(),
+                block_size: region.block_size().into(),
+                blocks_per_extent: region.blocks_per_extent(),
+                extent_count: region.extent_count(),
+                read_only: region.read_only(),
+            })
+            .collect();
+
+        let table = tabled::Table::new(rows)
+            .with(tabled::settings::Style::psql())
+            .to_string();
+
+        println!("{}", table);
+    }
+
+    Ok(())
+}
+
+/// Find what is using a region
+async fn cmd_db_region_used_by(
```

Comment:
This seems really cool, but.. does it just take any old UUID and...  how does it work?
Are we providing the UUID of the region in `RegionUsedByArgs`?

Reply:
We are yeah:
```
$ ./target/debug/omdb db region list
 id                                   | dataset_id                           | volume_id                            | block_size | blocks_per_extent | extent_count | read_only 
--------------------------------------+--------------------------------------+--------------------------------------+------------+-------------------+--------------+-----------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 3f7e5898-ec73-48bc-86a0-ef7a998b87db | bb7b9ada-9aab-4a9d-be89-23ddd09a29d2 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 42e49268-f0a7-431e-999e-3e82d03df045 | c728295a-97bc-4d42-bc31-f29b5c721208 | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 81f077e0-346d-46f8-bdd6-b2f84ad3f34e | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     
 90f98fe7-011b-4a16-9378-362d113ef245 | 70118760-7acd-4d8a-9fe7-7cd2d3c8e72d | ffdae556-6b03-4372-ba15-d2e8ad9fa1c1 | 512        | 131072            | 32           | false     
 a7725c22-cda9-4f30-b3dd-deadd94d812b | c728295a-97bc-4d42-bc31-f29b5c721208 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | 512        | 131072            | 32           | false     

$ ./target/debug/omdb db region used-by 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5
 id                                   | volume_id                            | usage_type | usage_id                             | usage_name           | deleted 
--------------------------------------+--------------------------------------+------------+--------------------------------------+----------------------+---------
 27eaaabc-ab7e-4d3a-bf54-e1bbc86384f5 | aa54c368-eee0-4e9b-86a9-58b1630da4bb | disk       | 1b39bc7d-2ef2-434d-a8eb-0149e437d6fe | crucible-tester-disk | false
 ```